src/dialogs/more-info/ha-more-info-tab-history.ts

```diff
@@ -27,78 +27,64 @@ export class MoreInfoTabHistoryDialog extends LitElement {
 
   @internalProperty() private _stateHistory?: HistoryResult;
 
-  @internalProperty() private _isLoading = false;
+  @internalProperty() private _entries?: LogbookEntry[];
 
-  private _entries: LogbookEntry[] = [];
-
-  private _persons = {};
+  @internalProperty() private _persons = {};
 
   private _historyRefreshInterval?: number;
 
   protected render(): TemplateResult {
     if (!this.entityId) {
       return html``;
     }
-    const entityId = this.entityId;
-    const stateObj = this.hass.states[entityId];
+    const stateObj = this.hass.states[this.entityId];
 
     if (!stateObj) {
       return html``;
     }
 
     return html`
-      ${this._isLoading
-        ? html`
-            <ha-circular-progress
-              active
-              alt=${this.hass.localize("ui.common.loading")}
-            ></ha-circular-progress>
-          `
-        : html`
-            <state-history-charts
-              up-to-now
-              .hass=${this.hass}
-              .historyData=${this._stateHistory}
-            ></state-history-charts>
-            <ha-logbook
-              narrow
-              no-click
-              no-icon
-              no-name
-              class=${classMap({ "no-entries": !this._entries.length })}
-              .hass=${this.hass}
-              .entries=${this._entries}
-              .userIdToName=${this._persons}
-            ></ha-logbook>
-          `}
+      <state-history-charts
+        up-to-now
+        .hass=${this.hass}
+        .historyData=${this._stateHistory}
+        .isLoadingData=${!this._stateHistory}
+      ></state-history-charts>
+      <ha-logbook
+        narrow
+        no-icon
+        no-name
+        class=${classMap({ "has-entries": Boolean(this._entries?.length) })}
+        .hass=${this.hass}
+        .entries=${this._entries || []}
+        .userIdToName=${this._persons}
+        .loading=${!this._entries}
+      ></ha-logbook>
```

Comment:
Wouldn't this be better?
```suggestion
      ${!this._entries ? html`<spinner></spinner>` : this._entries.length ? html`<ha-logbook
        narrow
        no-icon
        no-name
        .hass=${this.hass}
        .entries=${this._entries}
        .userIdToName=${this._persons}
        .loading=${!this._entries}
      ></ha-logbook>` : ""}
```

Reply:
Well the log book covers length of 0 entries

Reply:
I just mimicked the state history charts
but you are probably right that its better in the parent element

Reply:
I reset and forced pushed the changes